src/lib/db-listeners/db-student-personal-docs-listener.ts

```diff
@@ -0,0 +1,69 @@
+import firebase from "firebase/app";
+import { DB } from "../db";
+import { OtherDocumentType, PersonalDocument } from "../../models/document/document-types";
+import { BaseListener } from "./base-listener";
+
+export class DBStudentPersonalDocsListener extends BaseListener {
+  private db: DB;
+  private userPersonalDocsRefs: firebase.database.Reference[] = [];
+
+  constructor(db: DB, documentType: OtherDocumentType) {
+    super("DBStudentPersonalDocsListener");
+    this.db = db;
+  }
+
+  public start() {
+    const { user } = this.db.stores;
+    return new Promise<void>((resolve, reject) => {
+      const offeringUsersRef = this.db.firebase.ref(
+        this.db.firebase.getOfferingUsersPath(user)
+      );
+
+      const classPath = this.db.firebase.getClassPath(user);
+      this.debugLogHandler("#start", "adding", "once", offeringUsersRef);
+
+      offeringUsersRef.on("value", (snapshot) => {
+        const snapVal = snapshot.val();
+        if (!snapVal) return resolve();
+        const userKeys = Object.keys(snapVal).filter(key => key !== user.id);
+        const userPaths = userKeys.map(key => `${classPath}/users/${key}/personalDocs`);
+        const existingPaths = (this.userPersonalDocsRefs.map(ref => ref.toString()));
+
+        userPaths.forEach((userPath) => {
+          const potentiallyNewRef = this.db.firebase.ref(userPath);
+          if (!existingPaths.includes(potentiallyNewRef.toString())){
+            const ref = potentiallyNewRef;
+            this.debugLogHandlers("#start", "adding", ["child_added"], ref );
+            ref.on("child_added", this.handlePersonalDocAdded);
+            this.userPersonalDocsRefs.push(ref);
+          }
+        });
+        resolve();
+      });
+    });
+  }
+
+  public stop() {
+    if (this.userPersonalDocsRefs) {
+      this.userPersonalDocsRefs.forEach(ref => {
+        this.debugLogHandlers("#stop", "removing", ["child_added"], ref);
+        ref.off("child_added", this.handlePersonalDocAdded);
+      });
+    }
+  }
+
+  private handlePersonalDocAdded = (snapshot: firebase.database.DataSnapshot) => {
+    console.log("| path of snapshot ", snapshot.ref.toString());
```

Comment:
You could switch this to a `this.debugLogSnapshot(...` so it could be shown when the listeners debugging is enabled.